### ISSUE TITLE
floccus unsupported browser message(android)

### ISSUE THREAD

**Issue by swestlake** [floccusaddon/floccus#1000]
### Describe the bug
floccus attempts to sync(as an add-on to kiwi-browser on android), several minutes later says it cannot sync on unsupported browser.

### To Reproduce
1. Open kiwi browser and go to the chrome-extensions page of 'floccus'
2. install floccus extension
3. setup the bookmarks sync and wait for sync to complete

1. Use Android+Install Kiwi extension

### Expected behavior
green check-mark for a succeeded sync

### Screenshots
- no screenshots

### Desktop
(please complete the following information)
 - OS: Android
 - Browser Kiwi
 - Floccus version: 4.8.3
 - Floccus sync method: webdav

### Server
 - OS: linux
 - Owncloud version: 10.8 (webdav)

### Debug log
none

- [ ] Debug log provided? none

### Additional context
no additional context available

**Comment by marcelklehr** [comment]
Can you share a screenshot of the error, please?

**Comment by swestlake** [comment]
the sync runs for a long time with the 908 bookmark items..
(desktop syncs occur relatively quickly, however the android setup is
much longer that it is not really practical)

eventually the message becomes something along the lines that it is not
a support browser. It said something about the browser, and iirc it said unsupported.

I will try to give a screenshot at some point later today or tomorrow.

thanks

**Comment by marcelklehr** [comment]
Also see this issue about using floccus with owncloud: https://github.com/floccusaddon/floccus/issues/841

**Comment by swestlake** [comment]
@marcelklehr , take note the "Desktop" heading is misleading, and had to be used in order to fulfill the template requirements (for the bugreport).  The bug is taking place from android kiwi+floccus extension.

The sync is 100% successful only on Desktop web-browsers (four in total are used on multiple machines).

The only sync not working is the kiwi browser on Android.  No other Android browser has yet been tested, though I suppose those would be less compatible than kiwi at this currrent time according to notes on the site of the project.

the setting is set the same on all floccus instances(both desktop and android), so there's nothing differing in this...  so I would not be exactly sure as to why this isn't working.

I will try to check to see if there is a lock-file getting created (the hidden files would need be enabled on the webdav server as well).

**Comment by swestlake** [comment]
perhaps it could be valuable to troubleshoot the chrome-extension plugin, because all the settings are set the same on 5 web-browsers.

{{ 1 firefox, 1 chrome }}  on one machine
{{ 1 firefox, 1 chrome }}  on another machine on the same network

then
1 kiwi on Android.  Doesn't pass.

I am using the official "owncloud" docker image.

Question:: your chrome-extension should work with the official owncloud docker image as well?

something doesn't seem right....

**Comment by swestlake** [comment]
I think part of the reason why webdav is not working for some users is they are using the incorrect URL.
One symptom, is the connection stays on for a very long time and nothing happens... it's like stuck in a loop. << perhaps the logic in the extension needs to be improved for this.

"As far as I can remember webDAV sync with owncloud doesn't work"
^ you must be lacking testers in this area, because afaict webdav is working very stable here...and I am glad to be giving feedback!!

https://domain.com/remote.php/dav/files/user/  << is what I use for all my configurations.

I am using Owncloud 10.5 and 10.8... << floccus works on these two versions(docker)

On my first LAN, I am using owncloud 10.8 <<< works across 2 machines -- 4 desktop Web-browsers
On my second LAN, I am using owncloud 10.5 <<< works across 2 machines.  -- 2 desktop Web-browsers.

^^ The problem is NOT Desktop.
The problem is on the Android!!

So whatever code you're using for webdav for the Desktop-plugins, these afaict are 100% syncing and working correctly.

^ surprised !?

The problem is the SAME URL setting in the KIWI(Android) setup, is causing the red-error to show-up.

I will get to this later sometime to see if there is a tcpdump and lock-file getting made by the Android plugin.

**Comment by marcelklehr** [comment]
> ^^ The problem is NOT Desktop.
> The problem is on the Android!!

Are they using the same floccus version?

**Comment by marcelklehr** [comment]
Can you please post the exact error message here?

**Comment by swestlake** [comment]
> > ^^ The problem is NOT Desktop.
> > The problem is on the Android!!
> 
> Are they using the same floccus version?

same version 4.8.3

4.8.4 is not yet showing up on the chrome-extension shop..

**Comment by swestlake** [comment]
I am using another Android -- (and in another lan setup), and it passes with a simple bookmark sync. (only a few bookmarks in it)
I am not 100% sure what was happening with the other Android device..
I will try to replicate the red-error message as there was many bookmarks with that other setup.

**Comment by swestlake** [comment]
ran it on my latest setup --
A) 1 Desktop Chromium    (containing ~1000 bookmarks)
B) 1 Android Kiwi

Server: Owncloud 10.5,

Test 1:   No Bookmarks.xbel on server -- Create the bookmarks file on webdav server.
- Run first sync from (A)
(reminding to self -- I know from unmentioned coverage here is that for other desktop machines, they are able to sync with this newly created bookmarks.xbel without issue, so there is no need to consider these as being part of the test)

Test 2:  Test first-sync run on (B) (Kiwi on Android)
- Connects to server, and says "done"
I check new bookmarks in Kiwi, and there is nothing changed.
*** Despite Kiwi/Floccus saying success, there is zero new bookmarks showing in this app.  ***

Test 3: Delete Bookmarks.xbel from server, and re-run Test 1:, followed by Test 2:
^ This triggered a reset for Kiwi? --- apparently this did something for Kiwi to finally allow new bookmarks into it.

All tests say ok successful with a green checkmark, even though Test 2 is actually failing to show any new bookmark. (it fails to show any one new bookmark from a library bookmarks.xbel file containing a thousand bookmarks.)

There is a logic bug in the Android chrome-extension, as I can repeat the successful syncs...

I am not sure if I can replicate the "first-run" sync bug...

.. but I can replicate the "first-sync" bug presumably, if I completely wipe out the entire browser and user cache from Android, and start the whole thing over again.

^^ After the first-run failure (despite Kiwi/floccus saying "ok" successful), -- no matter how many times I would click on a manual sync, there would never be any new bookmarks.

(Whether this is irrelevant, maybe some food for thought::  I have zero bookmarks in Kiwi -- whether Kiwi comes with some default or not, I just delete any default bookmarks that may have been lingering when Kiwi browser is first installed. trivial, not important, though Kiwi remains at "0 bookmarks" continuously despite how many times I make changes to bookmarks.xbel from the Desktop browser.)

the Kiwi android plugin can clearly access and make the .lock, but it is using something incorrectly prevently it from adding new bookmarks.

Between Test 2: and Test 3: , I forced the Kiwi-Android plugin to create the new bookmarks.xbel file, deleted it from the server, then went to Test 3:.

^ The floccus extension as a brand new install should be "syncing bookmarks" on its first sync -- but it doesn't perform this (and again.. even though it says "successful" with the green checkmark).

..
The settings of the plugin, trying to use the default settings to keep things simple..
 -> **all settings are identical** between Desktop Browser and Kiwi/Android.

The settings differ from the application default in only-> URL, username, password and bookmarks file path.

the url I am testing is:
https://domain.com/remote.php/dav/files/_username/
(I am using a validated Lets Encrypt certificate, so Android doesn't give warnings on any self-signed certificates which aren't being used)

the bookmarks file path is:
"test/bookmarks.xbel"
-- and when a lock is noticed, this file appears as: "test/bookmarks.xbel.lock"

I joggled between Test 2: and Test 3: --- maybe once or twice... and the sync finally succeeds with "shown bookmarks" along with the standard green checkmark, etc.

so it is the first time I am noticing a successful sync on Android...

webdav looks very stable afaict, but I am not using anything like the specialized Nextcloud-Bookmarks service... it's just using the webdav file storage -- and seems to hold relatively well as I can see ".lock" added to the bookmarks.xbel filename whenever changes take place.

the webdav is owncloud 10.5..
the Android floccus extension is version 4.8.3, and 
the Desktop/Chromium floccus extension is also version 4.8.3

(maybe some of these issues have been addressed for v.4.8.4? -- last I checked it is not yet downloadable from the chrome-add-on store)

however it is very noticeable that the Desktop browsers sync get the new bookmarks from bookmarks.xbel on their very first sync....  no additional secondary manual syncs are needed -- so the Desktop browser's floccus add-on have a better sync built in them.

reflecting back around the time of ""Test 1:"" -- no matter how many times I would click on a "manual sync" in the Kiwi-floccus extension, there would never be new bookmarks added to the Kiwi browser -- even though there is always the green check-mark saying successful.  It really meant "0 bookmarks" were added(when in fact there are 1000 bookmarks residing in the 192K sized file on the webdav server) -- there clearly are bookmarks on the server, it is just that the floccus-extension has trouble determining these new bookmarks on its "first-run".  

On consecutive runs later(after deleting bookmarks.xbel repeatdly) --- with no changes to the floccus settings, the floccus extension is finally able to sync and ADD new bookmarks for Kiwi.

I sound a little repeated in the above observation, but I am doing this because there's lack of feedback for webdav -- surprisingly it is working quite well I might say.

thanks for letting me elaborate a little more on this.. hopefully I didn't try to create too many confusing observational notes, it is only what I can observe from the front-end interface.. of what is a relatively stable setup I have here..

not many people -- I'd say not enough people know that this can work relatively well on webdav -- surprises me as well... I thought it would be crashing a lot more than this..

so far it is pretty stable.

I have yet to use it as a daily driver, I just installed/tested it -- and I am of course sticking with the owncloud-official docker images only. I don't know why people do not resort to testing things with "official" docker images from their respective projects, and I think they should, as it will force everyone to test more of the same server setup.. 

(perhaps this project can also promote the idea of supporting official images as this also can make the troubleshooting a lot easier)

keep up the great work
thanks!!
